Zend/zend_smart_str.c

```diff
@@ -169,3 +169,37 @@ ZEND_API void ZEND_FASTCALL _smart_string_alloc(smart_string *str, size_t len)
 		str->c = erealloc2(str->c, str->a + 1, str->len);
 	}
 }
+
+ZEND_API void ZEND_FASTCALL smart_str_append_escaped_truncated(smart_str *str, zend_string *value, size_t length)
+{
+	smart_str_append_escaped(str, ZSTR_VAL(value), MIN(length, ZSTR_LEN(value)));
+
+	if (ZSTR_LEN(value) > length) {
+		smart_str_appendl(str, "...", sizeof("...")-1);
+	}
+}
+
+ZEND_API void ZEND_FASTCALL smart_str_append_scalar(smart_str *dest, zval *value, size_t truncate) {
+	ZEND_ASSERT(Z_TYPE_P(value) <= IS_STRING);
```

Comment:
I think this function would be better as a switch + EMPTY_SWITCH_DEFAULT_CASE().

Reply:
okay